File: Tests/CSharp/TestAllDatabases/TestAllDatabases/TestFixtures/MetadataFixture.cs
//======================================
// Copyright EntitySpaces, LLC 2005 - 2006 
//======================================

using System;
using System.Collections.Generic;
using System.Text;

using NUnit.Framework;
//using Adapdev.UnitTest;
using BusinessObjects;
using EntitySpaces.Interfaces;

namespace Tests.Base
{
	[TestFixture]
	public class MetadataFixture
	{
		AggregateTestCollection aggTestColl = new AggregateTestCollection();
		AggregateTest aggTest = new AggregateTest();
		AggregateTestQuery aggTestQuery = new AggregateTestQuery();

		[TestFixtureSetUp]
		public void Init()
		{
		}

		[SetUp]
		public void Init2()
		{
			aggTestColl = new AggregateTestCollection();
			aggTest = new AggregateTest();
			aggTestQuery = new AggregateTestQuery();
		}

		[Test]
		public void DescriptionWithDoubleQuote()
		{
			ComputedTest comp = new ComputedTest();

            switch (comp.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
//                  comp.AddNew();
                    string desc = comp.GetDescriptionDoubleQuoteTest();

                    Assert.AreEqual("See \"this\"", desc);
                    break;

                default:
                    Assert.Ignore("Tested for SQL Server only.");
                    break;
            }
		}

		[Test]
		public void DescriptionWithBackSlash()
		{
			ComputedTest comp = new ComputedTest();

            switch (comp.es.Connection.ProviderSignature.DataProviderName)
            {
                case "EntitySpaces.SqlClientProvider":
					string desc = comp.GetDescriptionBackSlashTest();

					Assert.AreEqual("See C:\\this", desc);
                    break;

                default:
                    Assert.Ignore("Tested for SQL Server only.");
                    break;
            }
		}

		[Test]
		public void GetColumnName()
		{
			AggregateTestMetadata meta = AggregateTestMetadata.Meta();

			foreach (esColumnMetadata col in meta.Columns)
			{
				string colName = col.Name;
				Assert.IsTrue(colName != string.Empty);
			}
		}

        [Test]
        public void GetColumnWithAliasEntity()
        {
            AggregateTest entity = new AggregateTest();
            entity.Query.es.CountAll = true;
            entity.Query.es.CountAllAlias = "Count";
            entity.Query.Load();

            Assert.AreEqual(30, Convert.ToInt32(entity.GetColumn("Count")));
        }

        [Test]
        public void GetColumnWithAliasEntityView()
        {
            FullNameView entity = new FullNameView();

            if (entity.es.Connection.Name == "SqlCe"
                || entity.es.Connection.ProviderMetadataKey ==
                "esSqlCe4"
                || entity.es.Connection.ProviderMetadataKey ==
                "esSqlCe")
            {
                Assert.Ignore("Not tested for SqlCe.");
            }
            else
            {
                entity.Query.es.CountAll = true;
                entity.Query.es.CountAllAlias = "Count";
                entity.Query.Load();

                Assert.AreEqual(16, Convert.ToInt32(entity.GetColumn("Count")));
            }
        }

        [Test]
        public void GetColumnWithAliasCollection()
        {
            AggregateTestCollection coll = new AggregateTestCollection();
            coll.Query.es.CountAll = true;
            coll.Query.es.CountAllAlias = "Count";
            coll.Query.Load();

            Assert.AreEqual(30, Convert.ToInt32(coll[0].GetColumn("Count")));
        }

        [Test]
		public void ChangeMetadata()
		{
			AggregateTest entity = new AggregateTest();

			Assert.IsTrue(entity.GetAutoKey() == true);
			entity.ToggleAutoKey();
			Assert.IsTrue(entity.GetAutoKey() == false);
			entity.ToggleAutoKey();
			Assert.IsTrue(entity.GetAutoKey() == true);
		}

        [Test]
        public void ConfigNotReadOnly()
        {
            AggregateTest entity = new AggregateTest();
            Assert.AreEqual("AggregateDb", entity.es.Connection.Name);

            string oldDefault = esConfigSettings.ConnectionInfo.Default;
            esConfigSettings.ConnectionInfo.Default = "ForeignKeyTest";

            entity = new AggregateTest();
            Assert.AreEqual("ForeignKeyTest", entity.es.Connection.Name);

            if (entity.es.Connection.ProviderSignature.DataProviderName == "EntitySpaces.OracleClientProvider")
            {
                Assert.IsTrue(entity.es.Connection.ConnectionString.ToLower().Contains("hierarchical"));
            }
            else
            {
                Assert.IsTrue(entity.es.Connection.ConnectionString.ToLower().Contains("foreignkeytest"));
            }

            entity.es.Connection.ConnectionString = "Test";
            Assert.AreEqual("Test", entity.es.Connection.ConnectionString);

            esConfigSettings.ConnectionInfo.Default = oldDefault;
        }
    }
}

File: Tests/VB/TestAllDatabases/TestAllDatabases/TestFixtures/MetadataFixture.vb
'======================================
' Copyright EntitySpaces, LLC 2005 - 2006 
'======================================

Imports System.Collections.Generic
Imports System.Text

Imports NUnit.Framework
'using Adapdev.UnitTest;
Imports BusinessObjects
Imports EntitySpaces.Interfaces

Namespace Tests.Base
	<TestFixture> _
	Public Class MetadataFixture
		Private aggTestColl As New AggregateTestCollection()
		Private aggTest As New AggregateTest()
		Private aggTestQuery As New AggregateTestQuery()

		<TestFixtureSetUp> _
		Public Sub Init()
		End Sub

		<SetUp> _
		Public Sub Init2()
			aggTestColl = New AggregateTestCollection()
			aggTest = New AggregateTest()
			aggTestQuery = New AggregateTestQuery()
		End Sub

		<Test> _
		Public Sub DescriptionWithDoubleQuote()
			Dim comp As New ComputedTest()

			Select Case comp.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					'                  comp.AddNew();
					Dim desc As String = comp.GetDescriptionDoubleQuoteTest()

					Assert.AreEqual("See ""this""", desc)
					Exit Select
				Case Else

					Assert.Ignore("Tested for SQL Server only.")
					Exit Select
			End Select
		End Sub

		<Test> _
		Public Sub DescriptionWithBackSlash()
			Dim comp As New ComputedTest()

			Select Case comp.es.Connection.ProviderSignature.DataProviderName
				Case "EntitySpaces.SqlClientProvider"
					Dim desc As String = comp.GetDescriptionBackSlashTest()

					Assert.AreEqual("See C:\this", desc)
					Exit Select
				Case Else

					Assert.Ignore("Tested for SQL Server only.")
					Exit Select
			End Select
		End Sub

		<Test> _
		Public Sub GetColumnName()
			Dim meta As AggregateTestMetadata = AggregateTestMetadata.Meta()

			For Each col As esColumnMetadata In meta.Columns
				Dim colName As String = col.Name
				Assert.IsTrue(colName <> String.Empty)
			Next
		End Sub

		<Test> _
		Public Sub GetColumnWithAliasEntity()
			Dim entity As New AggregateTest()
			entity.Query.es.CountAll = True
			entity.Query.es.CountAllAlias = "Count"
			entity.Query.Load()

			Assert.AreEqual(30, Convert.ToInt32(entity.GetColumn("Count")))
		End Sub

		<Test> _
		Public Sub GetColumnWithAliasEntityView()
			Dim entity As New FullNameView()

			If entity.es.Connection.Name = "SqlCe" OrElse entity.es.Connection.ProviderMetadataKey = "esSqlCe" Then
				Assert.Ignore("Not tested for SqlCe.")
			Else
				entity.Query.es.CountAll = True
				entity.Query.es.CountAllAlias = "Count"
				entity.Query.Load()

				Assert.AreEqual(16, Convert.ToInt32(entity.GetColumn("Count")))
			End If
		End Sub

		<Test> _
		Public Sub GetColumnWithAliasCollection()
			Dim coll As New AggregateTestCollection()
			coll.Query.es.CountAll = True
			coll.Query.es.CountAllAlias = "Count"
			coll.Query.Load()

			Assert.AreEqual(30, Convert.ToInt32(coll(0).GetColumn("Count")))
		End Sub

		<Test> _
		Public Sub ChangeMetadata()
			Dim entity As New AggregateTest()

			Assert.IsTrue(entity.GetAutoKey() = True)
			entity.ToggleAutoKey()
			Assert.IsTrue(entity.GetAutoKey() = False)
			entity.ToggleAutoKey()
			Assert.IsTrue(entity.GetAutoKey() = True)
		End Sub

		<Test> _
		Public Sub ConfigNotReadOnly()
			Dim entity As New AggregateTest()
			Assert.AreEqual("AggregateDb", entity.es.Connection.Name)

			Dim oldDefault As String = esConfigSettings.ConnectionInfo.[Default]
			esConfigSettings.ConnectionInfo.[Default] = "ForeignKeyTest"

			entity = New AggregateTest()
			Assert.AreEqual("ForeignKeyTest", entity.es.Connection.Name)

			If entity.es.Connection.ProviderSignature.DataProviderName = "EntitySpaces.OracleClientProvider" Then
				Assert.IsTrue(entity.es.Connection.ConnectionString.ToLower().Contains("hierarchical"))
			Else
				Assert.IsTrue(entity.es.Connection.ConnectionString.ToLower().Contains("foreignkeytest"))
			End If

			entity.es.Connection.ConnectionString = "Test"
			Assert.AreEqual("Test", entity.es.Connection.ConnectionString)

			esConfigSettings.ConnectionInfo.[Default] = oldDefault
		End Sub
	End Class
End Namespace
